src/main/java/com/minecolonies/coremod/inventory/ContainerItemHandler.java

```diff
@@ -0,0 +1,118 @@
+package com.minecolonies.coremod.inventory;
+
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.inventory.Container;
+import net.minecraft.inventory.IInventory;
+import net.minecraft.inventory.Slot;
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.items.IItemHandler;
+import net.minecraftforge.items.SlotItemHandler;
+
+import javax.annotation.Nullable;
+
+/**
+ * Created by asie on 2/16/17.
+ */
+public class ContainerItemHandler extends Container {
+    private final IItemHandler handler;
+    private final int numRows;
+
+    final int CONSTANT_INVENTORY_ROWS_PLAYER = 3;
+    final int CONSTANT_INVENTORY_COLUMNS     = 9;
+
+    public ContainerItemHandler(IInventory playerInventory, IItemHandler chestInventory, EntityPlayer player)
+    {
+        final int SLOT_SIZE = 18;
+
+        final int SLOT_X_OFFSET = 8;
+        final int SLOT_Y_OFFSET = 8;
+
+        final int PLAYER_SLOT_INDEX_OFFSET = 9;
+        final int PLAYER_SLOT_Y_OFFSET = 103;
+        final int PLAYER_HOTBAR_Y_OFFSET = 161;
+
+
+        this.handler = chestInventory;
+        this.numRows = chestInventory.getSlots() / CONSTANT_INVENTORY_COLUMNS;
+        int i = (this.numRows - 4) * 18;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Assign this magic number 4 to a well-named constant, and use the constant instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS109)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Assign this magic number 18 to a well-named constant, and use the constant instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS109)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Local variable 'i' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)